pex/pep_427.py

```diff
@@ -0,0 +1,310 @@
+            headers=sysconfig_paths["include"],
+            scripts=sysconfig_paths["scripts"],
+            data=sysconfig_paths["data"],
+        )
+
+    purelib = attr.ib()  # type: str
+    platlib = attr.ib()  # type: str
+    headers = attr.ib()  # type: str
+    scripts = attr.ib()  # type: str
+    data = attr.ib()  # type: str
+
+    def __getitem__(self, item):
+        # type: (Text) -> str
+        if "purelib" == item:
+            return self.purelib
+        elif "platlib" == item:
+            return self.platlib
+        elif "headers" == item:
+            return self.headers
+        elif "scripts" == item:
+            return self.scripts
+        elif "data" == item:
+            return self.data
+        raise KeyError("Not a known install path: {item}".format(item=item))
+
+
+def install_wheel_chroot(
+    wheel_path,  # type: str
+    destination,  # type: str
+    compile=False,  # type: bool
+    requested=True,  # type: bool
+):
+    # type: (...) -> InstalledWheel
+
+    metadata_files = install_wheel(
+        wheel_path,
+        InstallPaths.chroot(
+            destination,
+            project_name=ProjectNameAndVersion.from_filename(wheel_path).canonicalized_project_name,
+        ),
+        compile=compile,
+        requested=requested,
+    )
+
+    record_relpath = metadata_files.metadata_file_rel_path("RECORD")
+    assert (
+        record_relpath is not None
+    ), "The {module}.install_wheel function should always create a RECORD.".format(module=__name__)
+
+    direct_url_file_relpath = metadata_files.metadata_file_rel_path("direct_url.json")
+    if direct_url_file_relpath:
+        direct_url_file_abspath = os.path.join(
+            metadata_files.metadata.location, direct_url_file_relpath
+        )
+        if os.path.isfile(direct_url_file_abspath):
+            with open(direct_url_file_abspath) as fp:
+                if urlparse.urlparse(json.load(fp).get("url", "")).scheme == "file":
+                    os.unlink(direct_url_file_abspath)
+
+    return InstalledWheel.save(
+        prefix_dir=destination, stash_dir=InstallPaths.CHROOT_STASH, record_relpath=record_relpath
+    )
+
+
+def install_wheel_interpreter(
+    wheel_path,  # type: str
+    interpreter,  # type: PythonInterpreter
+    compile=True,  # type: bool
+    requested=True,  # type: bool
+):
+    # type: (...) -> MetadataFiles
+
+    return install_wheel(
+        wheel_path,
+        InstallPaths.interpreter(interpreter),
+        interpreter=interpreter,
+        compile=compile,
+        requested=requested,
+    )
+
+
+def install_wheel(
+    wheel_path,  # type: str
+    install_paths,  # type: InstallPaths
+    interpreter=None,  # type: Optional[PythonInterpreter]
+    compile=False,  # type: bool
+    requested=True,  # type: bool
+):
+    # type: (...) -> MetadataFiles
+
+    # See: https://packaging.python.org/en/latest/specifications/binary-distribution-format/#installing-a-wheel-distribution-1-0-py32-none-any-whl
```

Comment:
N.B.: This is pretty necessary reading to understand the spec that guides most of this code. It was originally encoded in PEP-427; thus the module name, but now resides here. I find the PyPA confusing.